packages/keystone/src/lib/core/queries/resolvers.ts

```diff
@@ -200,7 +200,7 @@ function applyMaxResults(results: unknown[], list: InitialisedList, context: Key
     throw limitsExceededError({ list: list.listKey, type: 'maxResults', limit: list.maxResults });
   }
   if (context) {
-    context.totalResults += Array.isArray(results) ? results.length : 1;
+    context.totalResults += results.length;
```

Comment:
(this is unrelated to the `take` thing, i just saw that this was always an array now)